spock-core/src/main/groovy/spock/util/EmbeddedSpecCompiler.groovy

```diff
@@ -82,17 +83,19 @@ class EmbeddedSpecCompiler {
     doCompile "package apackage; $imports ${source.trim()}"
   }
 
-  Class compileSpecBody(String source) {
+  Class compileSpecBody(@Language(value = 'Groovy', prefix = 'class ASpec extends spock.lang.Specification { ', suffix = '}')
```

Comment:
the suffix is missing a newline to be identical to the inner code, same goes for every declaration